src/pyqasm/preprocess.py

```diff
@@ -0,0 +1,80 @@
+# Copyright 2025 qBraid
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""
+Pre-processing prior to loading OpenQASM files as QasmModule objects.
+"""
+import os
+import re
+
+from pyqasm.exceptions import ValidationError
+
+
+def process_include_statements(program: str, filename: str) -> str:
+    """
+    Process include statements in a QASM file.
+
+    Args:
+        program (str): The QASM program string.
+        filename (str): Path to the QASM file (for resolving relative includes).
+
+    Returns:
+        str: The processed QASM program with includes injected.
+
+    Raises:
+        FileNotFoundError: If an include file is not found or cannot be read.
+    """
+    program_lines = program.splitlines()
+    processed_files = set()
+
+    for idx, line in enumerate(program_lines):
+        line = line.strip()
+        if line.startswith("include"):
+            # Extract include filename from quotes
+            match = re.search(r'include\s+["\']([^"\']+)["\']', line)
+            if not match:
+                raise ValidationError("Invalid include statement detected in QASM file.")
+            include_filename = match.group(1)
+            # Skip stdgates.inc and already processed files
+            if include_filename == "stdgates.inc" or include_filename in processed_files:
```

Comment:
I think we need to cater to `qelib1.inc` as well for qasm2 backward compatibility